### ISSUE TITLE
Flaky test: 24/10k: PEMFileProviderEnd2End

### ISSUE THREAD

**Issue by easwars** [grpc/grpc-go#5273]
https://github.com/grpc/grpc-go/runs/5685143497?check_suite_focus=true

**Comment by menghanl** [comment]
https://github.com/grpc/grpc-go/runs/5890181660?check_suite_focus=true

**Comment by zasweq** [comment]
https://github.com/grpc/grpc-go/runs/6477129401?check_suite_focus=true

**Comment by zasweq** [comment]
https://github.com/grpc/grpc-go/runs/6512136640?check_suite_focus=true

**Comment by easwars** [comment]
https://github.com/grpc/grpc-go/runs/7662833877?check_suite_focus=true

**Comment by easwars** [comment]
https://github.com/grpc/grpc-go/actions/runs/3200073817/jobs/5226556483

**Comment by arvindbr8** [comment]
Running a 10K run on this test doesn't seem be to flaking. 

`test results`
```
INFO: Elapsed time: 127.568s, Forge stats: 0/10000 actions cached, 48.8m CPU used, 1.4s queue time, 61.2 MB ObjFS output (novel bytes: 2.9 MB), 0.0 MB local output, Critical Path: 102.02s, Remote (99.24% of the time): [queue: 0.00%, setup: 0.18%, process: 0.62%, retry: 97.11%]
//third_party/golang/grpc/test:test_test                                 PASSED in 101.2s
  Stats over 10000 runs: max = 101.2s, min = 0.4s, avg = 2.1s, dev = 3.8s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
INFO: Streaming build results to: http://sponge2/31d24041-3914-4161-ae2a-0205412f3177
INFO: Build completed successfully, 10002 total actions
```

closing this one out. can be re-opened if it shows up

**Comment by easwars** [comment]
https://github.com/grpc/grpc-go/actions/runs/3987902725/jobs/6838333984

**Comment by arvindbr8** [comment]
Hi @erm-g, friendly ping. Let us know if we can help.

**Comment by rockspore** [comment]
This should also be fixed by #6093

**Comment by arjan-bal** [comment]
https://github.com/grpc/grpc-go/actions/runs/16284972768/job/45981803538?pr=8448
